verilog/analysis/verilog_linter.cc

```diff
@@ -77,12 +83,226 @@ using verible::LintWaiver;
 using verible::TextStructureView;
 using verible::TokenInfo;
 
+struct LintViolationWithStatus {
+  const verible::LintViolation* violation;
+  const verible::LintRuleStatus* status;
+
+  LintViolationWithStatus(const verible::LintViolation* v,
+                          const verible::LintRuleStatus* s)
+      : violation(v), status(s) {}
+
+  bool operator<(const LintViolationWithStatus& r) const {
+    // compares addresses which correspond to locations within the same string
+    return violation->token.text().data() < r.violation->token.text().data();
+  }
+};
+
+void LintStatusHandler::HandleLintRuleStatuses(
+    const std::vector<LintRuleStatus>& statuses, absl::string_view base,
+    absl::string_view path) {
+  std::set<LintViolationWithStatus> violations;
+
+  for (auto& status : statuses) {
+    for (auto& violation : status.violations) {
+      violations.insert(LintViolationWithStatus(&violation, &status));
+    }
+  }
+
+  for (auto violation : violations) {
+    HandleViolation(*violation.violation, base, path, violation.status->url,
+                    violation.status->lint_rule_name);
+  }
+}
+
+void ViolationPrinter::HandleLintRuleStatuses(
+    const std::vector<verible::LintRuleStatus>& statuses,
+    absl::string_view base, absl::string_view path) {
+  verible::LintStatusFormatter formatter(base);
+
+  formatter_ = &formatter;
+  LintStatusHandler::HandleLintRuleStatuses(statuses, base, path);
```

Comment:
Uhm, using inheritance to have some sort of 'common handling with decoration around' make for really hard to maintain code. Consider composition or having the `HandleLintRuleStatuses()` even be a free function.

Reply:
Refactored. https://github.com/google/verible/pull/668/commits/9f4ef1b89b783d929444a9f348113ba4a24caf14